### DISCUSSION TITLE
Adding path with "/" character via API

### DISCUSSION THREAD

**Discussion by robertmielewczyk** [bluenviron/mediamtx#580] [Questions]
# Which version are you using?
v0.17.3

# Which operating system are you using?
OS
- [x] linux
- [ ] windows
- [ ] mac
Architecture
- [x] amd64
- [ ] arm64
- [ ] arm7
- [ ] arm6

# Describe the issue
I found that it's possible to create a following config.yml with a fully working path url.
```yml
paths:
  path/with/slash:
```
response:
```
[0/0] [path path/with/slash] created 
```
The problem I'm having is that I'm not able to recreate this path via API. I've tried looking if there are any blocked characters in internal/core/api.go but I didn't find any line blocking name characters. Which leads me to believe that it's not possible to build such path when api url is: `/v1/config/paths/add/:name` as additional slashes make it so that it thinks it a different path.

## Here is what I tried and didn't solve my problem
```bash
# Works
curl -X POST -d {} -H "application/json" localhost:9997/v1/config/paths/add/video-path

# Doesn't Work (404 page not found)
curl -X POST -d {} -H "application/json" localhost:9997/v1/config/paths/add/path/with/slash
curl -X POST -d {} -H "application/json" localhost:9997/v1/config/paths/add/path%2Fwith%2Fslash

# 200 response but no path added
curl -X POST -d {} -H "application/json" localhost:9997/v1/config/paths/add/path%5Cwith%5Cslash

# Works but doesn't work when trying to connect a client (rtsp-simple-server error: path must end with slash)
curl -X POST -d {} -H "application/json" localhost:9997/v1/config/paths/add/path.with.dot.separator
```

Is there any way to make it work currently, or is it possible to change url from `/v1/config/paths/add` and add pathName into POST data. 

I think that unreserved list that is available here https://en.wikipedia.org/wiki/Percent-encoding describes which characters are currently possible to use when making a path

**Comment by aler9** [comment]
moved here #581
